pkg/noderesourcetopology/least_numa.go

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package noderesourcetopology
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/klog/v2"
+	v1qos "k8s.io/kubernetes/pkg/apis/core/v1/helper/qos"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/bitmask"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	"gonum.org/v1/gonum/stat/combin"
+	"sigs.k8s.io/scheduler-plugins/pkg/util"
+)
+
+var zeroQty resource.Quantity
+
+func init() {
+	zeroQty = resource.MustParse("0")
+}
+
+func leastNUMAContainerScopeScore(pod *v1.Pod, zones topologyv1alpha1.ZoneList) (int64, *framework.Status) {
+	nodes := createNUMANodeList(zones)
+	qos := v1qos.GetPodQOS(pod)
+
+	maxNUMANodesCount := 0
+	// the order how TopologyManager asks for hint is important so doing it in the same order
+	// https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/topologymanager/scope_container.go#L52
+	for _, container := range append(pod.Spec.InitContainers, pod.Spec.Containers...) {
+		identifier := fmt.Sprintf("%s/%s/%s", pod.Namespace, pod.Name, container.Name)
+		numaNodes, numaNodesCount := numaNodesRequired(identifier, qos, nodes, container.Resources.Requests)
```

Comment:
can `numaNodesCount` ever be different from `len(numaNodes.GetBits())` ?
I guess only on error. If so it could be worth to return `bitmask.Bitmask, error` WDYT?

Reply:
Totally makes sense 😄 my first attempt on this function was just to return `numaNodesCount` and after further refinements and adding `numaNodes` as return value I just didn't think of using it. Will change it now.

Reply:
changed.